crypto/fipsmodule/rsa/internal.h

```diff
@@ -69,6 +69,49 @@ extern "C" {
 
 typedef struct bn_blinding_st BN_BLINDING;
 
+struct rsa_meth_st {
+    void *app_data;
+
+    int (*init)(RSA *rsa);
+    int (*finish)(RSA *rsa);
+
+    // size returns the size of the RSA modulus in bytes.
+    size_t (*size)(const RSA *rsa);
+
+    int (*sign)(int type, const uint8_t *m, unsigned int m_length,
+                uint8_t *sigret, unsigned int *siglen, const RSA *rsa);
+
+    // Set via |RSA_meth_set_priv_enc|
+    int (*sign_raw)(int max_out, const uint8_t *in, uint8_t *out, RSA *rsa,
+                    int padding);
+    // Set via |RSA_meth_set_pub_dec|
+    int (*verify_raw)(int max_out, const uint8_t *in, uint8_t *out, RSA *rsa,
+                      int padding);
+    // Set via |RSA_meth_set_priv_dec|
+    int (*decrypt)(int max_out, const uint8_t *in, uint8_t *out, RSA *rsa,
+                   int padding);
+    // Set via |RSA_meth_set_pub_enc|
+    int (*encrypt)(int max_out, const uint8_t *in, uint8_t *out, RSA *rsa,
+                   int padding);
```

Comment:
The names are a little misleading. Private key encryption and public key decryption aren't really "normal operations" for RSA keys, sign_raw and verify_raw are clearer in indicating what the operation is doing. Since the function pointers are part of an internal struct, the names don't really matter - I picked ones that are clearer to understand. We can always change this if we decide to make the struct public and want to maintain OpenSSL compatibility - I've documented this in a clearer way.